### PR TITLE
Test customer invoicing with do_not_invoice

### PR DESCRIPTION
**Pull request by gbova**

### CODE REVIEW

**Comment by nickpell** [comment]
Test failure looks related:

```
FAIL: corehq.apps.accounting.tests.test_domain_user_history:TestDomainUserHistory.test_calculate_users_in_all_domains
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/accounting/tests/test_domain_user_history.py", line 38, in test_calculate_users_in_all_domains
    self.assertEqual(DomainUserHistory.objects.count(), 1)
AssertionError: 3 != 1
```

File: corehq/apps/accounting/tests/test_customer_invoicing.py [hunk 1]
```diff
@@ -180,15 +190,9 @@ def test_product_line_items(self):
         invoice = CustomerInvoice.objects.first()
         product_line_items = invoice.lineitem_set.get_products()
         self.assertEqual(product_line_items.count(), 2)
-        for line_item in product_line_items:
-            self.assertIn(line_item.base_description, [
-                'One month of CommCare Advanced Edition Software Plan.',
-                'One month of CommCare Standard Edition Software Plan.',
-            ])
-            self.assertIn(line_item.base_cost, [
```
Comment (nickpell) [inline corehq/apps/accounting/tests/test_customer_invoicing.py]: Looks like test coverage is being removed re: ```line_item.base_cost```.  Is that ok/desirable?

File: corehq/apps/accounting/tests/test_customer_invoicing.py [hunk 2]
```diff
@@ -180,15 +190,9 @@ def test_product_line_items(self):
...
-                self.product_rate.monthly_fee,
-                self.advanced_plan.product_rate.monthly_fee,
-            ])
+        product_descriptions = [line_item.base_description for line_item in product_line_items]
+        self.assertIn('One month of CommCare Advanced Edition Software Plan.', product_descriptions)
```
Comment (nickpell) [inline corehq/apps/accounting/tests/test_customer_invoicing.py]: Nit: I wonder if using ```self.assertItemsEqual``` would be clearer here.

File: corehq/apps/accounting/tests/test_customer_invoicing.py (hunk 1 above)
Comment (gbova) [reply corehq/apps/accounting/tests/test_customer_invoicing.py]: Nope. That's definitely a mistake

**Review by nickpell — APPROVED** [review]

**Pull request merged**